XMonad/Util/ClickableWorkspaces.hs

```diff
@@ -32,26 +33,33 @@ import XMonad.Hooks.DynamicLog (xmobarAction, xmobarRaw, PP(..))
 -- > logHook = clickablePP xmobarPP { ... } >>= dynamicLogWithPP
 --
 -- * Requirements:
---   * wmctrl on system (in path)
+--   * @xdotool@ on system (in path)
 --   * "XMonad.Hooks.EwmhDesktops" for wmctrl support (see Hackage docs for setup)
 --   * use of UnsafeStdinReader in xmobarrc (rather than StdinReader)
 
 
 clickableWrap :: Int -> String -> String
-clickableWrap i ws = xmobarAction ("wmctrl -s " ++ show i) "1" $ xmobarRaw ws
+clickableWrap i ws = xmobarAction ("xdotool set_desktop " ++ show i) "1" $ xmobarRaw ws
 
 -- Use index of workspace in users config to target workspace with wmctrl switch.
-getClickable :: X (WorkspaceId -> String)
-getClickable = do
+getClickable :: (WorkspaceId -> String) -> X (WorkspaceId -> String)
```

Comment:
If this is really a "renaming" function, then it should be `WorkspaceId -> WorkspaceId`, right?

Reply:
I like it more this way, as it conveys the idea that the input should really be a valid workspace name (as opposed to something that's already been renamed) whereas the output can be anything at all.

Reply:
Aha okay, that makes sense